### PR TITLE
Wip rgw reshard old bucket

### PR DESCRIPTION
**Pull request by oritwas**

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -12987,6 +12988,10 @@ int RGWRados::cls_user_sync_bucket_stats(rgw_raw_obj& user_obj, const RGWBucketI
       entry.size += header_stats.total_size;
       entry.size_rounded += header_stats.total_size_rounded;
       entry.count += header_stats.num_entries;
+      r = check_bucket_shards(bucket_info, bucket_info.bucket, bucket_quota);
+      if (r < 0) {
+	ldout(cct, 0) << "ERROR in check_bucket_shards: " << cpp_strerror(-r)<< dendl;
+      }
```
Comment (yehudasa) [inline src/rgw/rgw_rados.cc]: @oritwas shouldn't have this inside the loop here, need to move it outside. Actually, I think this is the wrong method to put this in, maybe `rgw_user.cc:rgw_user_sync_all_stats()` is better?

**Comment by oritwas** [comment]
@yehudasa , fixed please review

**Comment by oritwas** [comment]
@yehudasa , my test:
1. created  bucket with 10 objs 
2 . rerun the gateway with --rgw-max_objs-per-shard=8 --rgw-user-quota-sync-wait-time=5 --rgw-reshard-thread-interval =5
3. check it was put into the reshard queue
4. check the number of the bucket shard is 2 (after it was resharded)

**Comment by oritwas** [comment]
jenkins test this please

**Pull request merged**